Makefile

```diff
@@ -151,19 +144,49 @@ test-ext-dev-e2e: $(OPERATOR_SDK) $(KUSTOMIZE) $(KIND) #HELP Run extension creat
 	test/extension-developer-e2e/setup.sh $(OPERATOR_SDK) $(CONTAINER_RUNTIME) $(KUSTOMIZE) $(KIND) $(KIND_CLUSTER_NAME) $(E2E_REGISTRY_NAMESPACE)
 	go test -count=1 -v ./test/extension-developer-e2e/...
 
-.PHONY: test-unit
+# By default, setup-envtest will write to $XDG_DATA_HOME or $HOME/.local/share if neither is set.
+# If $HOME is unset, specify a binary directory to prevent an error in setup-envtest.
+# This is useful for CI/CD environments without $XDG_DATA_HOME or $HOME.
+LOCALBIN := $(PWD)/bin
+SETUP_ENVTEST := $(LOCALBIN)/setup-envtest
+SETUP_ENVTEST_BIN_DIR_OVERRIDE := --bin-dir $(LOCALBIN)/envtest-binaries
+
+ifeq ($(shell [[ -z "$$HOME" || "$$HOME" == "/" ]] && [[ -z "$$XDG_DATA_HOME" ]] && echo true), true)
+	SETUP_ENVTEST := /tmp/setup-envtest
```

Comment:
I am not sure if we are really using this part
but the binary used to setup-envtest is either downloaded.